breadcord/core_modules/auto_update/__init__.py

```diff
@@ -78,27 +84,64 @@ async def update_modules(self) -> None:
                     await git('rev-parse', 'HEAD', cwd=module.path)
                 ):
                     self.logger.debug(f'Module {module.id} is up-to-date.')
-                    return
+                    continue
                 self.logger.warning(f'Module {module.id} is out-of-date.')
-                await self.update_module(module)
+                pull_msg, commit_hash, commit_msg = await self.update_module(module)
+                updated_modules[module.id] = pull_msg, commit_hash, commit_msg
             except subprocess.CalledProcessError as error:
                 self.logger.error(f'Failed to fetch updates for the module {module.id!r}: {error}\n{error.stderr}')
 
-    async def update_module(self, module: breadcord.module.Module) -> None:
+        if updated_modules:
+            self.logger.info(f'Finished updating len({updated_modules}) modules')
+        else:
+            self.logger.debug('No modules were updated')
+        return updated_modules
+
+    async def update_module(self, module: breadcord.module.Module) -> tuple[str, str, str]:
         self.logger.info(f'Updating {module.id}')
         update_text = await git('pull', cwd=module.path)
         self.logger.debug(f'({module.id}) Git output:\n{update_text.strip()}')
         await module.reload()
 
-        git_hash_msg = await git('log', '-1', '--format="%H %s"', cwd=module.path)
-        self.logger.debug(f'Module {module.id} now on: {git_hash_msg[1:-2]}')
+        git_hash_msg = (await git('log', '-1', '--format="%H %s"', cwd=module.path)).strip().strip('"')
+        self.logger.debug(f'Updated {module.id} to {git_hash_msg}')
+
+        parts = git_hash_msg.strip().split(' ', 1)
+        return update_text, parts[0], ' '.join(parts[1:])
 
     @commands.command()
     @commands.is_owner()
     async def update(self, ctx: commands.Context) -> None:
         message = await ctx.send('Updating...')
-        await self.update_modules()
-        await message.edit(content='Finished updating modules.')
+        updated = await self.update_modules()
+        if len(updated) == 0:
+            await message.edit(content='No modules were updated.')
+            return
+
+        embeds = [
+            discord.Embed(
+                title=f'Updated module `{module_id}`',
+                description='\n'.join(filter(bool, (
+                    f'**Latest commit message**: {discord.utils.escape_markdown(commit_msg)}',
+                    f'```\n{pull_msg}\n```' if pull_msg else None,
```

Comment:
_Technically_ this could go stinky if the pull messages contains triple backticks but like, I've seen the most popular bots out there trip up on this and I dunno if it's worth trying to fix

Reply:
That's what I was thinking, yep - although how would that deal with consecutive chains of more backticks?

Reply:
It does not really need to, does it?

Reply:
No idea! Not sure how Discord behaves with that